nats-concepts/jetstream/consumers.md

```diff
@@ -1,234 +1,91 @@
+## Push-specific
 
-### DeliverByStartSequence
+These options apply only to push consumers. For an example on how to configure a push consumer using your preferred client, see [NATS by Example](https://natsbyexample.com/examples/jetstream/push-consumer/go).
 
-When first consuming messages, start at this particular message in the set. The consumer is required to specify `OptStartSeq`, the sequence number to start on. It will receive the closest available message moving forward in the sequence should the message specified have been removed based on the stream limit policy.
-
-### DeliverByStartTime
-
-When first consuming messages, start with messages on or after this time. The consumer is required to specify `OptStartTime`, the time in the stream to start at. It will receive the closest available message on or after that time.
-
-## Durable \(Name\)
-
-The name of the Consumer, which the server will track, allowing resuming consumption where left off. By default, a consumer is ephemeral. To make the consumer durable, set the name.
-
-{% hint style="warning" %}
-The consumer's name **_can not_** be adjusted after the initial creation of the consumer
-{% endhint %}
-
-## Description
-
-A description of the Consumer. This is particularly useful for ephemeral consumers to indicate their purpose since the durable name cannot be provided.
-
-{% hint style="info" %}
-The consumer's description _can_ be adjusted after the initial creation of the consumer
-{% endhint %}
+| Field | Description | Version | Editable |
+| :--- | :--- | :--- | :--- |
+| DeliverSubject | The subject to deliver messages to. Note, setting this field implicitly decides whether the consumer is push or pull-based. With a deliver subject, the server will _push_ messages to client subscribed to this subject. | 2.2.0 | No |
+| DeliverGroup | The queue group name which, if specified, is then used to distribute the messages between the subscribers to the consumer. This is analogous to a [queue group][queue-group] in core NATS. | 2.2.0 | Yes |
+| FlowControl | Enables per-subscription flow control using a sliding-window protocol. This protocol relies on the server and client exchanging messages to regulate when and how many messages are pushed to the client. This flow control mechanism is in addition to the flow control imposed by `MaxAckPending` across all subscriptions bound to a consumer. | 2.2.0 | Yes |
+| IdleHeartbeat | If the idle heartbeat period is set, the server will regularly send a status message to the client (i.e. when the period has elapsed) while there are no new messages to send. This lets the client know that the JetStream service is still up and running, even when there is no activity on the stream. The message status header will have a code of 100. Unlike FlowControl, it will have no reply to address. It may have a description like "Idle Heartbeat". | 2.2.0 | Yes |
```

Comment:
Maybe say that this handling of the heartbeat is done by the client library, doesn't have to be done by the developper?